src/Ocelot/Logging/OcelotDiagnosticListener.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DiagnosticAdapter;
+
+namespace Ocelot.Logging
+{
+    public class OcelotDiagnosticListener
+    {
+        [DiagnosticName("Microsoft.AspNetCore.MiddlewareAnalysis.MiddlewareStarting")]
+        public virtual void OnMiddlewareStarting(HttpContext httpContext, string name)
+        {
+            Console.WriteLine($"MiddlewareStarting: {name}; {httpContext.Request.Path}");

```

Comment:
I might have missed something here but shouldnt this be taking in an ILoggerFactory and building a logger to output these messages rather than writing to the console?